File: client-app/src/app/models/event.ts
import { Profile } from "./profile";
import { Field } from "./field";

export interface Event {
  id: string;
  title: string;
  date: string;
  description: string;
  category: string;
  city: string;
  maximumAttendees: number;
  isMaximumAttendensReached: boolean;
  hostUserName?: string;
  isCanceled: boolean;
  isHost: boolean;
  isGoing: boolean;
  host?: Profile;
  attendees: Profile[];
  fields: Field[];
}

export class Event implements Event {
  constructor(init?: EventFormValues) {
    Object.assign(this, init);
  }
}

export class EventFormValues {
  id?: string = undefined;
  title: string = "";
  category: string = "";
  description: string = "";
  date: string = "";
  city: string = "";
  maximumAttendees?: number;
  fields: Field[] = [];

  constructor(event?: EventFormValues) {
    if (event) {
      this.id = event.id;
      this.title = event.title;
      this.category = event.category;
      this.description = event.description;
      this.date = event.date;
      this.city = event.city;
      this.maximumAttendees = event.maximumAttendees;
      this.fields = event.fields;
    }
  }
}

File: Domain/Event.cs
namespace Domain;
public class Event
{
public Guid Id { get; set; }
public string Title { get; set; }
public DateTime Date { get; set; }
public string Description { get; set; }
public string Category { get; set; }    
public string City { get; set; }    
public int MaximumAttendees { get; set; }
public bool IsCanceled { get; set; }
public ICollection<EventAttendee> Attendees { get; set; } = new List<EventAttendee>();
public ICollection<Field> Fields { get; set; } = new List<Field>();
public ICollection<Comment> Comments { get; set; } = new List<Comment>();
}
